lib/faraday/adapter/net_http.rb

```diff
@@ -89,11 +89,16 @@ def with_net_http_connection(env)
       end
 
       def net_http_connection(env)
-        if proxy = env[:request][:proxy]
-          Net::HTTP::Proxy(proxy[:uri].host, proxy[:uri].port, proxy[:user], proxy[:password])
-        else
-          Net::HTTP
-        end.new(env[:url].host, env[:url].port || (env[:url].scheme == 'https' ? 443 : 80))
+        proxy = env[:request][:proxy]
```

Comment:
`env[:request][:proxy]` could be nil, which causes all `proxy[...]` calls to fail later in this case.
You will need to test it with `if proxy.nil?` and use the correct `Net::HTTP.new` method accordingly